nixos/lib/make-loader-options.nix

```
@@ -0,0 +1,38 @@
+{ lib }:
+/* Build a bootloader set of options */
+let
+  inherit (lib) mkOption types mdDoc;
+
+  commonOptions = {
+      id = mkOption {
+        type = types.str;
+        description = mdDoc "Identifier of this external bootloader for metadata";
+      };
+
+      supportsInitrdSecrets = mkOption {
+        type = types.bool;
+        default = false;
+        internal = true;
+        description = mdDoc ''
+         Whether the bootloader setup runs append-initrd-secrets.
+         If not, any needed secrets must be copied into the initrd
+         and thus added to the store.
+
+         This means that the "secrets" are absolutely not secrets
+         with respect to that feature.
```

Hm? What do you mean here?

I think the alternative is that if it's not supported it's not supported. This sounds like it would implicitly add secrets in an insecure way.

```suggestion
         When unsupported, and initrdSecrets are configured to be used,
         the system build will error out accordingly.
```

Or am I misunderstanding?